apps/lib/win32_init.c

```diff
@@ -8,300 +8,97 @@
-            case L'\\':
-                /*
-                 * Microsoft documentation on how backslashes are treated
-                 * is:
-                 *
-                 * + Backslashes are interpreted literally, unless they
-                 *   immediately precede a double quotation mark.
-                 * + If an even number of backslashes is followed by a double
-                 *   quotation mark, one backslash is placed in the argv array
-                 *   for every pair of backslashes, and the double quotation
-                 *   mark is interpreted as a string delimiter.
-                 * + If an odd number of backslashes is followed by a double
-                 *   quotation mark, one backslash is placed in the argv array
-                 *   for every pair of backslashes, and the double quotation
-                 *   mark is "escaped" by the remaining backslash, causing a
-                 *   literal double quotation mark (") to be placed in argv.
-                 *
-                 * Ref: https://msdn.microsoft.com/en-us/library/17w5ykft.aspx
-                 *
-                 * Though referred page doesn't mention it, multiple qouble
-                 * quotes are also special. Pair of double quotes in quoted
-                 * string is counted as single double quote.
-                 */
-                {
-                    const WCHAR *q = p;
-                    int i;
-
-                    while (*p == L'\\')
-                        p++;
-
-                    if (*p == L'"') {
-                        int i;
-
-                        for (i = (p - q) / 2; i > 0; i--)
-                            *wend++ = L'\\';
-
-                        /*
-                         * if odd amount of backslashes before the quote,
-                         * said quote is part of the argument, not a delimiter
-                         */
-                        if ((p - q) % 2 == 1)
-                            *wend++ = *p++;
-                    } else {
-                        for (i = p - q; i > 0; i--)
-                            *wend++ = L'\\';
-                    }
-                }
-                break;
-            case L'"':
-                /*
-                 * Without the preceding backslash (or when preceded with an
-                 * even number of backslashes), the double quote is a simple
-                 * string delimiter and just slightly change the parsing state
-                 */
-                if (in_quote && p[1] == L'"')
-                    *wend++ = *p++;
-                else
-                    in_quote = !in_quote;
-                p++;
-                break;
-            default:
-                /*
-                 * Any other non-delimiter character is just taken verbatim
-                 */
-                *wend++ = *p++;
-            }
-        }
-
-        wlen = wend - warg;
+    cmd_args = CommandLineToArgvW(cmd_line_args, &argc);
```

Comment:
You get build errors for using this...  [the MSVC manual](https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-commandlinetoargvw) tells us that you must link with shell32.lib to get that function.  That library should be added here:

https://github.com/openssl/openssl/blob/dd73b45e289e02d4fa9c7be0c72581b833ac9219/Configurations/10-main.conf#L1598